src/components/tiles/table/table-tile.tsx

```diff
@@ -211,19 +209,27 @@ const TableToolComponent: React.FC<ITileProps> = observer(function TableToolComp
     return () => disposer();
   });
 
+  // Currently this is recreated on each render, so the ToolbarContext is changed
+  // on each render. deleteSelected is changed on each render, so a useMemo
+  // here would not help the situation. I think an object that is managing
+  // the internal state of the component would be a better way to factor
+  // all of the use* calls above.
+  const toolbarContext = {
+    showExpressionsDialog,
+    deleteSelected,
+  };
```

Comment:
`useDataSet()` should really be returning stable functions, e.g. by using `useCallback()`, and then `useMemo()` here would have the desired effect.

Reply:
I find all of the use* calls in this component really hard to follow. I'm not sure but I think there should be a way to continue to have the features of the component broken off into parts but make it easier to follow. I was thinking something like an object that is built by composing all of the parts. And then this object can provide all of the properities that need to be passed to the RDG.